### ISSUE TITLE
Remove/Add node bug in simple.DirectedGraph

### ISSUE THREAD

**Issue by jtwatson** [gonum/graph#123]
In the implementation of simple.DirectedGraph.RemoveNode() there is a bug that causes NewNodeID() to panic.  Here is an example that will cause the panic:

```
package main

import (
	"math"

	"github.com/gonum/graph/simple"
)

func main() {
	g := NewDirectedGraph(0, math.Inf(1))

	n0 := Node(g.NewNodeID())
	g.AddNode(n0)

	n1 := Node(g.NewNodeID())
	g.AddNode(n1)

	g.RemoveNode(n0)

	n2 := Node(g.NewNodeID())
	g.AddNode(n2)

}
```

```
panic: simple: node ID collision: 2

goroutine 1 [running]:
github.com/gonum/graph/simple.(*DirectedGraph).AddNode(0xc820045d20, 0x4a1318, 0xc82000ad90)
	/go/src/github.com/gonum/graph/simple/directed.go:73 +0x1b8
```

I have a simple pull request to fix this, but if you would want a more robust fix to deal with Node ID wrap around with better performance, let me know.

**Comment by kortschak** [comment]
Nice find. Thank you. The free list should hold n0's ID though. What is happening instead?

**Comment by kortschak** [comment]
There is an issue with your case, you need to interleave the NewNodeID and AddNode calls.

**Comment by jtwatson** [comment]
I went ahead and interleaved all NewNodeID and AddNode calls, but due to NewNodeID() incrementing g.MaxID, I don't see a functional difference.  Maybe I miss-understood your comment.

**Comment by kortschak** [comment]
Yes, sorry I could be more clear. It doesn't fix the problem here, but it is incorrect use of the API. If you get two node IDs and try to add them both, you will get a collision since getting an ID does not change the state of the graph.

**Comment by kortschak** [comment]
Fixed by #124.
